cumulusci/core/utils.py

```diff
@@ -151,15 +155,123 @@ def decode_to_unicode(content):
 
 
 def merge_config(configs):
-    """recursively deep-merge the configs into one another (highest priority comes first)"""
-    new_config = {}
+    """
+    First remove any flow steps that are being overridden so that there are no conflicts
+    or different step types after merging. Then recursively deep-merge the configs into
+    one another (highest priority comes first)
+    """
+    configs = cleanup_flow_step_override_conflicts(configs)
```

Comment:
We might want to deepcopy `configs` first to make sure that we don't mutate the structures that were passed in. I forget the scenario but I remember there were some bugs in the past that came from dictmerge not being careful to copy on write.

Reply:
done.